src/EtherFiAdmin.sol

```diff
@@ -171,28 +175,18 @@ contract EtherFiAdmin is Initializable, OwnableUpgradeable, UUPSUpgradeable {
     }
 
     // Process `liquidityPool.batchApproveRegistration` for the last task in the pending queue
-    function AdminTask_ApproveValidators(bytes32 _reportHash, uint256[] memory _validatorsToApprove, bytes[] calldata _pubKey, bytes[] calldata _signature) external isAdmin() {
-        AdminTask memory task = _dequeuePendingTask("validatorsToApprove");
-
+    function AdminTask_ApproveValidatorTask(bytes32 _reportHash, uint256[] memory _validatorsToApprove, bytes[] calldata _pubKey, bytes[] calldata _signature) external isAdmin() {
         require(etherFiOracle.isConsensusReached(_reportHash), "EtherFiAdmin: report didn't reach consensus");
-        require(task.operationHash == keccak256(abi.encode(_reportHash, _validatorsToApprove, _pubKey, _signature)), "EtherFiAdmin: invalid task");
+        require(ValidatorApprovalTaskStatus[keccak256(abi.encode(_reportHash, _validatorsToApprove))].exists, "EtherFiAdmin: task doesn't exist");
 
         liquidityPool.batchApproveRegistration(_validatorsToApprove, _pubKey, _signature);
+        ValidatorApprovalTaskStatus[keccak256(abi.encode(_reportHash, _validatorsToApprove))].completed = true;
```

Comment:
not an issue here, but better to find the convention
https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html

so, updating `.completed` flag should come first before `liquidityPool.batchApproveRegistration(`